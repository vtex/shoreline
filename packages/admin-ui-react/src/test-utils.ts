import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
