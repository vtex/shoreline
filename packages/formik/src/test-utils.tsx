import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
