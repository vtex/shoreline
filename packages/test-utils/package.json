{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
