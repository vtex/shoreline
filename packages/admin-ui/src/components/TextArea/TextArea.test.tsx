import React, { Fragment } from 'react'
import { render, axe } from '../../test-utils'

import { TextArea } from './index'

describe('TextArea tests', () => {
  it('should not have a11y violations', async () => {
    const { container } = render(
      <Fragment>
        <TextArea
          charLimit={120}
          helperText="Helper Text"
          criticalText="Error Message"
          value=""
          onChange={() => {}}
          label="Label"
          id="text-area-1"
        />
        <TextArea
          charLimit={120}
          helperText="Helper Text"
          criticalText="Error Message"
          value=""
          onChange={() => {}}
          tone="critical"
          label="Label"
          id="text-area-2"
        />
        <TextArea
          charLimit={120}
          helperText="Helper Text"
          value=""
          onChange={() => {}}
          tone="critical"
          label="Label"
          disabled
          id="text-area-3"
        />
      </Fragment>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
