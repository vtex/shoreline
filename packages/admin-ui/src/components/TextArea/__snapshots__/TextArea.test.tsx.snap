// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.emotion-4 input:focus+label,
.emotion-4 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-4 input:placeholder-shown:not(:focus)+label,
.emotion-4 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-4 input:not(:placeholder-shown)+label,
.emotion-4 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  padding-top: 24px;
  height: 100px;
  resize: none;
  width: 100%;
  padding-left: 0.75rem;
  padding-right: 1rem;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  color: #000000;
  background: transparent;
  border: 1px solid #B9B9B9;
}

.emotion-0:disabled {
  background: #E6E6E6;
  color: #747474;
  border: 1px solid #747474;
}

.emotion-0:not(:focus):hover {
  border: 1px solid #545454;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

<div
    class="emotion-4"
  >
    <textarea
      class="emotion-0"
      id="text-area-1"
      placeholder=" "
    />
    <label
      class="emotion-1"
      for="text-area-1"
    >
      Label
    </label>
    <div
      class="emotion-3"
    >
      <span
        class="emotion-2"
      >
        Helper Text
      </span>
    </div>
  </div>
  .emotion-4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.emotion-4 input:focus+label,
.emotion-4 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-4 input:placeholder-shown:not(:focus)+label,
.emotion-4 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-4 input:not(:placeholder-shown)+label,
.emotion-4 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  padding-top: 24px;
  height: 100px;
  resize: none;
  width: 100%;
  padding-left: 0.75rem;
  padding-right: 1rem;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  color: #000000;
  background: transparent;
  border: 1px solid #B9B9B9;
}

.emotion-0:disabled {
  background: #E6E6E6;
  color: #747474;
  border: 1px solid #747474;
}

.emotion-0:not(:focus):hover {
  border: 1px solid #545454;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

<div
    class="emotion-4"
  >
    <textarea
      class="emotion-0"
      id="text-area-2"
      maxlength="120"
      placeholder=" "
    />
    <label
      class="emotion-1"
      for="text-area-2"
    >
      Label
    </label>
    <div
      class="emotion-3"
    >
      <div />
      <span
        class="emotion-2"
      >
        0/120
      </span>
    </div>
  </div>
  .emotion-4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.emotion-4 input:focus+label,
.emotion-4 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-4 input:placeholder-shown:not(:focus)+label,
.emotion-4 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-4 input:not(:placeholder-shown)+label,
.emotion-4 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  padding-top: 24px;
  height: 100px;
  resize: none;
  width: 100%;
  padding-left: 0.75rem;
  padding-right: 1rem;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  color: #000000;
  background: form.critical;
  border: 1px solid #CC3D3D;
}

.emotion-0:disabled {
  background: #E6E6E6;
  color: #747474;
  border: 1px solid #747474;
}

.emotion-0:not(:focus):hover {
  border: 1px solid #962828;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #FBF2F2,0rem 0rem 0rem 0.1875rem #F49494;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #CC3D3D;
}

<div
    class="emotion-4"
  >
    <textarea
      class="emotion-0"
      id="text-area-3"
      placeholder=" "
    />
    <label
      class="emotion-1"
      for="text-area-3"
    >
      Label
    </label>
    <div
      class="emotion-3"
    >
      <span
        class="emotion-2"
      >
        Critical Message
      </span>
    </div>
  </div>
  .emotion-5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.emotion-5 input:focus+label,
.emotion-5 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-5 input:placeholder-shown:not(:focus)+label,
.emotion-5 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-5 input:not(:placeholder-shown)+label,
.emotion-5 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  padding-top: 24px;
  height: 100px;
  resize: none;
  width: 100%;
  padding-left: 0.75rem;
  padding-right: 1rem;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  color: #000000;
  background: transparent;
  border: 1px solid #B9B9B9;
}

.emotion-0:disabled {
  background: #E6E6E6;
  color: #747474;
  border: 1px solid #747474;
}

.emotion-0:not(:focus):hover {
  border: 1px solid #545454;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

<div
    class="emotion-5"
  >
    <textarea
      class="emotion-0"
      id="text-area-4"
      maxlength="120"
      placeholder=" "
    />
    <label
      class="emotion-1"
      for="text-area-4"
    >
      Label
    </label>
    <div
      class="emotion-4"
    >
      <span
        class="emotion-2"
      >
        Helper Text
      </span>
      <span
        class="emotion-2"
      >
        0/120
      </span>
    </div>
  </div>
  .emotion-5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.emotion-5 input:focus+label,
.emotion-5 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-5 input:placeholder-shown:not(:focus)+label,
.emotion-5 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-5 input:not(:placeholder-shown)+label,
.emotion-5 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  padding-top: 24px;
  height: 100px;
  resize: none;
  width: 100%;
  padding-left: 0.75rem;
  padding-right: 1rem;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  color: #000000;
  background: form.critical;
  border: 1px solid #CC3D3D;
}

.emotion-0:disabled {
  background: #E6E6E6;
  color: #747474;
  border: 1px solid #747474;
}

.emotion-0:not(:focus):hover {
  border: 1px solid #962828;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #FBF2F2,0rem 0rem 0rem 0.1875rem #F49494;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #CC3D3D;
}

<div
    class="emotion-5"
  >
    <textarea
      class="emotion-0"
      id="text-area-5"
      maxlength="120"
      placeholder=" "
    />
    <label
      class="emotion-1"
      for="text-area-5"
    >
      Label
    </label>
    <div
      class="emotion-4"
    >
      <span
        class="emotion-2"
      >
        Critical Message
      </span>
      <span
        class="emotion-3"
      >
        0/120
      </span>
    </div>
  </div>
</DocumentFragment>
`;
